pulp_ostree/tests/functional/api/test_modify.py

```diff
@@ -1,275 +1,270 @@
-
-        self.repo2 = self.repositories_api.read(self.repo2.pulp_href)
-        self.assertEqual(self.repo2.latest_version_href, f"{self.repo2.pulp_href}versions/1/")
-
-        repository_version = self.versions_api.read(self.repo2.latest_version_href)
-        added_content = repository_version.content_summary.added
-        self.assertEqual(added_content["ostree.refs"]["count"], 1)
-        self.assertEqual(added_content["ostree.commit"]["count"], 2)
-        self.assertEqual(added_content["ostree.object"]["count"], 3)
-
-        distribution_data = OstreeOstreeDistribution(
-            **gen_distribution(repository=self.repo2.pulp_href)
-        )
-        response = self.distributions_api.create(distribution_data)
-        distribution = monitor_task(response.task).created_resources[0]
-        self.addCleanup(self.distributions_api.delete, distribution)
-
-        ostree_repo_path = self.distributions_api.read(distribution).base_url
-
-        remote_name = init_local_repo_with_remote(self.remote.name, ostree_repo_path)
-        self.addCleanup(shutil.rmtree, self.remote.name)
-        validate_repo_integrity(
-            self.remote.name,
-            f"{remote_name}:{ref1.name}",
-            commits_to_check={latest_commit.checksum},
-            depth=0,
-        )
-
-    def test_copy_whole_repository(self):
-        """Initialize a new repository from the existing repository."""
-        self.repo2 = self.repositories_api.create(gen_repo())
-        self.addCleanup(self.repositories_api.delete, self.repo2.pulp_href)
 
-        modify_repo(self.cfg, self.repo2.to_dict(), base_version=self.repo_version1.pulp_href)
 
-        self.repo2 = self.repositories_api.read(self.repo2.pulp_href)
-        self.assertEqual(self.repo2.latest_version_href, f"{self.repo2.pulp_href}versions/1/")
-
-        self.repo1 = self.repositories_api.read(self.repo1.pulp_href)
-        repo1_content = normalize_content(get_content(self.repo1.to_dict()))
-        repo2_content = normalize_content(get_content(self.repo2.to_dict()))
+@pytest.fixture
+def normalize_content(
+    ostree_content_commits_api_client,
+    ostree_content_configs_api_client,
+    ostree_content_objects_api_client,
+    ostree_content_refs_api_client,
+    ostree_content_summaries_api_client,
+):
+    """Make the repository content to be ready for comparison purposes."""
 
-        self.assertEqual(repo1_content, repo2_content, repo2_content)
+    def _normalize_content(repo):
+        commits = ostree_content_commits_api_client.list(repository_version=repo).results
+        configs = ostree_content_configs_api_client.list(repository_version=repo).results
+        objects = ostree_content_objects_api_client.list(repository_version=repo).results
+        refs = ostree_content_refs_api_client.list(repository_version=repo).results
+        summaries = ostree_content_summaries_api_client.list(repository_version=repo).results
```

Comment:
Is there a guarantee that the results are always returned in the same order?

We used `sorted` in the past to enforce the order:
```
sorted(
        [remove_created_key(item) for item in repo_content[OSTREE_OBJECTS_NAME]],
        key=lambda item: item["pulp_href"],
    )
```

Reply:
hum... the only test that uses the `normalize_content` is the *test_copy_whole_repository* and, if I understood it right, the test will not create multiple contents (commits/configs/objects/refs/summaries), therefore the results will always be ordered (a list with 0 or one element is an ordered list), right?

Anyway, I'll order it so that we can use it in the future with other tests if necessary.

Reply:
The synced repository should contain 3 objects as far as I know.

Reply:
ohhh ...my bad! (I still need to study and learn ostree to understand what is going on)